Source/DafnyCore/Generic/Lazy.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+
+namespace Microsoft.Dafny;
+
+public class Lazy<T> {
```

Comment:
C# already provides a full-feature generic Lazy<> class
https://learn.microsoft.com/en-us/dotnet/api/system.lazy-1.-ctor?view=net-7.0
Can you please indicate the difference here?

Reply:
Thanks! Removed the custom implementation